File: app/src/main/java/wangdaye/com/geometricweather/location/BaiduIPLocationApi.java
package wangdaye.com.geometricweather.location;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;
import wangdaye.com.geometricweather.weather.json.location.BaiduIPLocationResult;

public interface BaiduIPLocationApi {

    @GET("location/ip")
    Observable<BaiduIPLocationResult> getLocation(@Query("ak") String ak,
                                                  @Query("coor") String coor);
}

File: app/src/main/java/wangdaye/com/geometricweather/location/apis/BaiduIPLocationApi.kt
package wangdaye.com.geometricweather.location.apis

import io.reactivex.rxjava3.core.Observable
import retrofit2.http.GET
import retrofit2.http.Query
import wangdaye.com.geometricweather.location.json.BaiduIPLocationResult

interface BaiduIPLocationApi {
    @GET("location/ip")
    fun getLocation(
        @Query("ak") ak: String,
        @Query("coor") coor: String
    ): Observable<BaiduIPLocationResult>
}
